plugins/web/opentelemetry-plugin-react-load/src/BaseOpenTelemetryComponent.ts

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import { isWrapped } from '@opentelemetry/core';
+import * as shimmer from 'shimmer';
+import { GeneralAttribute } from '@opentelemetry/semantic-conventions';
+import { AttributeNames } from './enums/AttributeNames';
+import * as React from 'react';
+import { VERSION } from './version';
+import {
+  RenderFunction,
+  ComponentDidMountFunction,
+  ComponentDidUpdateFunction,
+  ShouldComponentUpdateFunction,
+  SetStateFunction,
+  ForceUpdateFunction,
+  GetSnapshotBeforeUpdateFunction,
+  ComponentWillUnmountFunction
+} from './types';
+
+interface ParentSpanData {
+  span: api.Span,
+  name: string,
+}
+
+const DEFAULT_PARENT_NAME = 'parent';
+/**
+ * This class is the base component for a React component with lifecycle instumentation
+ */
+export class BaseOpenTelemetryComponent extends React.Component {
+  readonly component: string = 'react-load';
+  readonly version: string = '1';
+  moduleName = this.component;
+  private _parentSpanMap: WeakMap<React.Component, ParentSpanData>;
+  private static _tracer: api.Tracer;
+  private static _logger: api.Logger;
+  
+  /**
+   * @param props Props of the React component
+  */
+  constructor(props: any) {
+    super(props);
+    this._parentSpanMap = new WeakMap<React.Component, ParentSpanData>();
+    this.patch();
+  }
+
+  /**
+   * Sets the tracer for all components being instrumented
+   * @param name Name of tracer
+   * @param version Version of tracer, this is optional. When not provided it will use the latest.
+   */
+  static setTracer(name: string, version?: string){
+    this._tracer = api.trace.getTracer(name, version ? version : VERSION)
```

Comment:
minor suggestion: it Appears like you refer to the static member `_tracer` as `BaseOpenTelemetryComponent._tracer` throughout the rest of the file. Consider renaming this one as well to keep consistent, WDYT?